This patch fixes format string abuses in minicom.  There are likely
many more.
                         - Mike A. Harris <mharris@redhat.com>
			 
--- minicom-1.83.1/man/minicom.1.format-string-vuln	Thu May  3 20:14:00 2001
+++ minicom-1.83.1/man/minicom.1	Thu May  3 20:44:52 2001
@@ -166,7 +166,7 @@
 user creates his own configuration file, it will show up in his home
 directory as '.minirc.dfl'.
 .SH USE
-Minicom is windows-based. To popup a window with the function you
+Minicom is window based. To popup a window with the function you
 want, press Control-A (from now on, we will use C-A to mean
 Control-A), and then the function key (a-z or A-Z). By pressing C-A
 first and then 'z', a help screen comes up with a short summary of all
@@ -754,26 +754,8 @@
 Turkish is under construction.
 .PD 1
 .SH "SECURITY ISSUES"
-Since Minicom is run setuid root on some computers, you probably want to 
-restrict access to it. This is possible by using a configuration file 
-in the same directory as the default files, called "minicom.users". 
-The syntax of this file is as following:
-.PP
-.RS 0.5i
-<username> <configuration> [configuration...]
-.RE
-.PP
-To allow user 'miquels' to use the default configuration, enter the 
-following line into "minicom.users":
-.PP
-.RS 0.5i
-miquels dfl
-.RE
-.PP
-If you want users to be able to use more than the default
-configurations, just add the names of those configurations behind the
-user name. If no configuration is given behind the username, minicom
-assumes that the user has access to all configurations.
+Do not run minicom setuid root, or setgid anything.  It is not safe.  See
+the \fBBUGS\fP section for details.
 .PD 1
 .SH MISC
 If minicom is hung, kill it with SIGTERM . (This means kill -15, or
@@ -818,6 +800,14 @@
 .fi
 .SH VERSION
 Minicom is now up to version 1.83.1.
+.SH BUGS
+Minicom was NOT designed with security in mind.  As a result, it is chock 
+full of format string abuse bugs.  The do_log() function for example is 
+used by the log command in minicom's script language.
+The format string input is unchecked, allowing the possibility of a user
+elevating their priveledges to that of root if minicom is setuid root,
+or group rights if ran setgid.  There are likely many other exploitable
+bugs in minicom also if it is ran setuid/setgid.  You have been warned!
 .SH AUTHORS
 The original author of minicom is Miquel van Smoorenburg (miquels@cistron.nl).
 He wrote versions up to 1.75.
--- minicom-1.83.1/src/minicom.c.format-string-vuln	Thu May  3 18:38:30 2001
+++ minicom-1.83.1/src/minicom.c	Thu May  3 18:38:30 2001
@@ -1298,7 +1298,7 @@
 			vt_set(addlf, -1, NULL, -1, -1, -1, -1, -1);
 			s = addlf ?
 			  _("Add linefeed ON") : _("Add linefeed OFF");
-			werror(s);
+			werror("%s", s);
 			break;
 #if _SELECT
 		case 'e': /* Local echo on/off. */
@@ -1306,7 +1306,7 @@
 			vt_set(-1, -1, NULL, -1, -1, local_echo, -1 ,-1);
 			s = local_echo ?
 			  _("Local echo ON") : _("Local echo OFF");
-			werror(s);
+			werror("%s", s);
 			break;
 #endif
 		case 'z': /* Help */
@@ -1421,7 +1421,7 @@
 			c = (!us->wrap);
 			vt_set(addlf, c, capfp, docap, -1, -1, -1, -1);
 			s = c ? _("Linewrap ON") : _("Linewrap OFF");
-			werror(s);
+			werror("%s", s);
 			break;
 		case 'o': /* Configure Minicom */
 			(void) config(0);
--- minicom-1.83.1/src/updown.c.format-string-vuln	Thu Mar 16 04:44:26 2000
+++ minicom-1.83.1/src/updown.c	Thu May  3 18:38:30 2001
@@ -59,7 +59,7 @@
   	/* This may look safe but you might I8N change the string! so
   	   snprintf it */
   	snprintf(err, sizeof(err),  _("Cannot chdir to %.30s"), dir);
-  	werror(err);
+  	werror("%s", err);
   	return(-1);
   }
   return(0);
--- minicom-1.83.1/src/file.c.format-string-vuln	Thu May  3 19:48:02 2001
+++ minicom-1.83.1/src/file.c	Thu May  3 19:48:24 2001
@@ -77,7 +77,7 @@
 {
   WIN *w;
 
-  w = mc_tell(s);
+  w = mc_tell("%s", s);
   sleep(1);
   wclose(w, 1);
 }
--- minicom-1.83.1/src/dial.c.format-string-vuln	Thu May  3 19:33:38 2001
+++ minicom-1.83.1/src/dial.c	Thu May  3 19:40:47 2001
@@ -906,11 +906,11 @@
     if (snprintf(copycmd,sizeof(copycmd),
 	     "cp %s %s.%hd",dfile,dfile,dial_ver.size) > 0) {
       if (P_LOGFNAME[0] != 0)
-	do_log(copycmd);
+	do_log("%s", copycmd);
       if (system(copycmd) == 0) {
 	snprintf(copycmd,sizeof(copycmd),
 		 _("Old dialdir copied as %s.%hd"),dfile,dial_ver.size);
-	w=mc_tell(copycmd);
+	w=mc_tell("%s", copycmd);
 	if (w) {
 	  sleep(2);
 	  wclose(w,1);
